velox/functions/prestosql/StringFunctions.h

```diff
@@ -507,4 +507,78 @@ struct StrLPosFunction : public StrPosFunctionBase<T, true> {};
 template <typename T>
 struct StrRPosFunction : public StrPosFunctionBase<T, false> {};
 
+// Helper method to return a byte value from the buffer.
+template <typename T>
+inline T getByte(const char* buffer, const int32_t i) {
+  return (T)buffer[i] & 0xff;
+}
+
+template <typename T>
+struct FromBigEndian32 {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE
+  void call(out_type<int32_t>& result, const arg_type<Varbinary>& input) {
+    static constexpr auto kTypeLength = sizeof(int32_t);
+    VELOX_USER_CHECK_EQ(input.size(), kTypeLength, "Expected 4-byte input");
+    const char* buffer = input.data();
+    // Convert the big endian byte buffer into a little endian integer.
+    // Most of the platforms have little endian byte order so the following
+    // conversion will be ignored.
+    auto little_endian_result = getByte<int32_t>(buffer, 0) << 24 |
+        getByte<int32_t>(buffer, 1) << 16 | getByte<int32_t>(buffer, 2) << 8 |
+        getByte<int32_t>(buffer, 3);
+    // Swap in case the host byte order is big endian.
+    result = folly::Endian::little(little_endian_result);
```

Comment:
Can it be just

```c++
memcpy(&result, buffer, sizeof(int32_t));
result = folly::Endian::big(result);
```

Reply:
Oh, I did not know this could be done this way 😄. Let me update.

Reply:
I have updated the code with your suggestion.